### ISSUE TITLE
window.getComputedStyled doesn't work with v5.2.0 and Testing Library/JSDOM

### ISSUE THREAD

**Issue by dfernandez79** [styled-components/styled-components#3262]
## System:
 - OS: macOS 10.15.6
 - CPU: (8) x64 Intel(R) Core(TM) i7-8569U CPU @ 2.80GHz
 - Memory: 70.66 MB / 16.00 GB
 - Shell: 5.7.1 - /bin/zsh

## Binaries:
 - Node: 12.16.1 - ~/.nvm/versions/node/v12.16.1/bin/node
 - Yarn: 1.22.4 - ~/.yarn/bin/yarn
 - npm: 6.14.8 - ~/.nvm/versions/node/v12.16.1/bin/npm

## Reproduction

https://github.com/dfernandez79/styledcomp-bug-example

## Steps to reproduce

1. Clone the project: https://github.com/dfernandez79/styledcomp-bug-example
2. Install the dependencies: `npm ci`
3. Run the tests: `npm test`

## Expected Behavior

The tests pass with `v5.1.1` but fail with `v5.2.0`. The test is using `window.getComputedStyle` to get the styles. While the implementation of computed styles is limited in JSDOM, it worked in `v5.1.1` so I expect it to work in `v5.2.0`.

## Actual Behavior

With `v5.2.0` calls to `window.getComputedStyle` always return an empty string. Doing a `console.log(prettyDOM(document))` doesn't show any `<style>` tag for `v5.2.0`.

**Comment by Marcelolynx** [comment]
i get exact problem also using .toHaveStyle, in 5.1.1 version test pass but 5.2 no. i follow this thread.

**Comment by fooloomanzoo** [comment]
`toBeVisible` is not working for me in `testing-library`

```
      15 |     );
      16 | 
    > 17 |     expect(getByText('Item 1')).not.toBeVisible();
         |                                     ^
      18 |     fireEvent.click(getByText('Click to open'));
      19 |     expect(getByText('Item 1')).toBeVisible();
      20 |   });
```

**Comment by ryota-murakami** [comment]
@fooloomanzoo Yeah, [`toBeVisible` using `getComputedStyle` internal ](https://github.com/testing-library/jest-dom/blob/master/src/to-be-visible.js#L6) so might be that's something broken.

## This PR is my reproduction resource: [Update styled-components: 5.1.1 → 5.2.0 (minor) ](https://github.com/laststance/create-react-app-typescript-todo-example-2020/pull/284)

I have OSS project [Create React App TypeScript Todo Example 2020](https://github.com/laststance/create-react-app-typescript-todo-example-2020) and I also facing issue about `toBeVisible` of [@testing-library/jest-dom](https://github.com/testing-library/jest-dom) custom jest matchers.(it's always return true).


- **[Parent Commit(Not Broken)](https://github.com/laststance/create-react-app-typescript-todo-example-2020/tree/ce70d83f4ade543ce38bb3034ddbb944c6f57791)** of **[Update styled-components: 5.1.1 → 5.2.0 (minor)(Broken)](https://github.com/laststance/create-react-app-typescript-todo-example-2020/pull/284/commits/374621d1cf9ae721df1036ef68c099f05d059b8c)**

[Parent Commit CI result](https://github.com/laststance/create-react-app-typescript-todo-example-2020/runs/1075077639?check_suite_focus=true) is all green, but [Update styled-components to version 5.2.0 Commit CI result](https://github.com/laststance/create-react-app-typescript-todo-example-2020/runs/1075656205) showed break `yarn test`(jest + react-testing-library) at the `not.toBeVisible()` assertion.

<img width="887" alt="Screen Shot 2020-09-13 at 15 26 13" src="https://user-images.githubusercontent.com/5501268/93012045-c9c31300-f5d7-11ea-82d3-50f29bb9ac43.png">

**Comment by jnielson94** [comment]
For what it's worth, it looks like it worked through `5.2.0-test.7` (at least in my app). If I had to guess it's something with removing browser code from server environments (since I think JSDOM uses the server bundle?) - https://github.com/styled-components/styled-components/commit/2214485f9cf5f0e0ca6b7caffd93351b2e4130d3

**Comment by ryota-murakami** [comment]
@jnielson94 RTL render in jsdom, is styled-components apply server bundle in the jsdom?
[RTL using `ReactDOM.render()` ](https://github.com/testing-library/react-testing-library/blob/693228ce10f23e2b695730ea88dbdaa35506e00e/src/pure.js#L63) as long as you don't specify `hydrate` option.

**Comment by 615349** [comment]
why it is closed without any solution? 
I get the same issue that `window.getComputedStyle` stopped working after bumping the version from `5.1.1` to `5.2.0`

**Comment by michaelyuen** [comment]
> why it is closed without any solution?
> I get the same issue that `window.getComputedStyle` stopped working after bumping the version from `5.1.1` to `5.2.0`

This issue is not closed. You may be confused with the PR in React-Avancado being closed.

**Comment by gregberge** [comment]
The problem is that Jest does not use the `browser` field, if you don't want to break everything, you can just use that trick:

```
moduleNameMapper: {
    'styled-components':
      '<rootDir>/node_modules/styled-components/dist/styled-components.browser.cjs.js',
  },
```

**Comment by willianjusten** [comment]
@gregberge that works fine for cases using `toHaveStyle` or `toBeDisabled` but inside the snapshots we're losing the styles that comes from `jest-styled-components`....

**Comment by tbergq** [comment]
The solution from @gregberge works fine for me as well. 

@willianjusten try using 

```js
module.exports = {
  moduleNameMapper: {
    '^styled-components':
      '<rootDir>/node_modules/styled-components/dist/styled-components.browser.cjs.js',
  },
}
```

without the `^` it will also map `jest-styled-components` to `styled-components.browser.cjs.js`

**Comment by willianjusten** [comment]
Thanks @tbergq , it's working perfectly now =D

**Comment by gregberge** [comment]
I think it should be mentioned in docs, @tbergq has the final solution!

**Comment by johanarnor** [comment]
It seems like [v5.2.1](https://github.com/styled-components/styled-components/releases/tag/v5.2.1) have solve it (for us at least).

**Comment by dfernandez79** [comment]
I've updated the test app used for the bug report to `v5.2.1`: https://github.com/dfernandez79/styledcomp-bug-example

And the bug seems to be solved 5.2.1 solved there too. Closing this issue 😄

**Comment by DanielMarkiel** [comment]
@dfernandez79 It looks like the bug reappeared in the styled-component v6. It blocked me after upgrading to v6.0.3.
The solution posted by @tbergquist-godaddy solved the issue in my case.

> The solution from @gregberge works fine for me as well.
> 
> @willianjusten try using
> 
> ```js
> module.exports = {
>   moduleNameMapper: {
>     '^styled-components':
>       '<rootDir>/node_modules/styled-components/dist/styled-components.browser.cjs.js',
>   },
> }
> ```
> 
> without the `^` it will also map `jest-styled-components` to `styled-components.browser.cjs.js`
